### ISSUE TITLE
Template import complains about string indices must be integers

### ISSUE THREAD

**Issue by Zablove** [ansible-collections/community.zabbix#1113]
##### SUMMARY

During the initial of several templates there is no problem, however if the playbook ansible task is running again and the template already exists in Zabbix, the import fails. 

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

community.zabbix.zabbix_template

##### ANSIBLE VERSION

```paste below
2.14.11 & 2.15.5
```

##### CONFIGURATION

```paste below
No special config
```

##### OS / ENVIRONMENT / Zabbix Version

Tested with Ansible 2.14.11 (@Fedora) & 2.15.5 (@Ubuntu) against (upgraded) Zabbix 6.4.4 and fresh install of 6.4.7
Tested with the template that comes with zabbix "Systemd by Zabbix agent 2"

##### STEPS TO REPRODUCE

Execute the task:

- Export template "Systemd by Zabbix agent 2" to XML
- Rename "zbx_export_templates.xml" to "Systemd_by_Zabbix_agent_2.xml"
- Run task below:
```
- name: Import templates from repo XML
  community.zabbix.zabbix_template:
    template_xml: "{{ lookup('file', 'Systemd_by_Zabbix_agent_2.xml' ) }}"
    state: present
```
- Import will be succesfull
- Second run will fail with 
```
The full traceback is:
  File "/tmp/ansible_community.zabbix.zabbix_template_payload_hzagaubm/ansible_community.zabbix.zabbix_template_payload.zip/ansible_collections/community/zabbix/plugins/module_utils/api_request.py", line 29, in _httpapi_error_handle
    code, response = self.connection.send_request(data=payload)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/ansible_community.zabbix.zabbix_template_payload_hzagaubm/ansible_community.zabbix.zabbix_template_payload.zip/ansible/module_utils/connection.py", line 200, in __rpc__
    raise ConnectionError(to_text(msg, errors='surrogate_then_replace'), code=code)
fatal: [localhost -> x.x.x.x]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "clear_templates": null,
            "http_login_password": null,
            "http_login_user": null,
            "link_templates": null,
            "macros": null,
            "state": "present",
            "tags": null,
            "template_groups": null,
            "template_json": null,
            "template_name": null,
            "template_xml": "<?xml version= ....
            ..................... Output omitted .........
"msg": "connection error occurred: string indices must be integers, not 'str'"
```

Steps also tested with the same template but with export and import in json format, issue the same.
Steps also tested with the template "F5 Big-IP SNMP" (provided by default installation of Zabbix)

```yaml
- name: Import templates from repo XML
  community.zabbix.zabbix_template:
    template_xml: "{{ lookup('file', 'Systemd_by_Zabbix_agent_2.xml' ) }}"
    state: present
```

##### EXPECTED RESULTS

Expected an ok or changed (in case the template was changed).

##### ACTUAL RESULTS

No import of the template. Seems the job is failing at comparison of the current template at the Zabbix Server and the template in Ansible.

**Comment by BGmot** [comment]
Can't reproduce. First delete the template from clean Zabbix 6.4.8 installation.
Then create the template with your code:
```
ey@ls1:~/ansible$ ansible-playbook -i localhost, test_template_import.yml

PLAY [all] ******************************************************************************************************************************************************************************************************************************************************************

TASK [Import templates from repo XML] ***************************************************************************************************************************************************************************************************************************************
changed: [localhost]

PLAY RECAP ******************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```
Re-run again - nothing changed:
```
ey@ls1:~/ansible$ ansible-playbook -i localhost, test_template_import.yml

PLAY [all] ******************************************************************************************************************************************************************************************************************************************************************

TASK [Import templates from repo XML] ***************************************************************************************************************************************************************************************************************************************
ok: [localhost]

PLAY RECAP ******************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```
no errors, I can re-run as many times as I want.
Similar issue was fixed by https://github.com/ansible-collections/community.zabbix/pull/1058 if you on collection version v2.2.0 then it should not happening to you.

**Comment by Zablove** [comment]
Thanks for testing this, indeed solved in v2.2.0
In my understanding I was running the latest version because I had tested on a fresh install. However the version at Ansible Galaxy is still pointing to v2.1.0. After installing v2.2.0 from this git repo the issue was resolved.

**Comment by BGmot** [comment]
Sorry for inconvenience, 2.2.0 now in Ansible Galaxy.
